scala/scala.bzl

```diff
@@ -289,23 +289,65 @@ DependencyAnalyzerMode: {dependency_analyzer_mode}
         arguments=["--jvm_flag=%s" % flag for flag in ctx.attr.scalac_jvm_flags] + ["@" + argfile.path],
       )
 
+    java_jar = try_to_compile_java_jar(ctx, all_srcjars, java_srcs, implicit_junit_deps_needed_for_java_compilation)
+    return java_jar
 
-def _compile_or_empty(ctx, jars, srcjars, buildijar, transitive_compile_jars, jars2labels):
+
+def try_to_compile_java_jar(ctx, all_srcjars, java_srcs, implicit_junit_deps_needed_for_java_compilation):
+    if not java_srcs and not all_srcjars:
+      return False
+
+    providers_of_dependencies = collect_java_providers_of(ctx.attr.deps)
+    providers_of_dependencies += collect_java_providers_of(implicit_junit_deps_needed_for_java_compilation)
+    scala_sources_java_provider = java_common.create_provider(
+        #the line below means we can't use the provider from java_common.compile
+        #since it will include the full jar when I want to expose only the ijar
+        #on the other hand we can't supply the java_common.compile the scala ijar since it needs the internals
```

Comment:
I don't know what `internals` mean here. I'm not sure why the ijar of the scalac compilation along with all the ijars we need for compilation is not sufficient. Can you add to the comment to possibly explain this?

Reply:
Sure but maybe let's first see I'm not just mistaken :) : Won't ijar strip parts from the scalac jar the java sources need? They're the same compilation "group"

Reply:
Maybe a test can be devised for this...

Reply:
ijar strips the code and leaves the classes and the signatures. Imagine all methods were replaced with `throw new Exception("")`. I don't see how javac can use that code at compile time except as annotation processors (which are the closest things java has to macros as far as I know).

So, I think using the ijar is exactly what we want since bazel is now driving this compilation. There could be times where I think bazel could recompile scala, but see that the java didn't change and the compiled ijar didn't change.

Reply:
Sounds good. I'll use the ijar.
I wonder if I should add a no recompilation for same target when scala
changes and java doesn't recompile.
I'll probably add it

Reply:
👍 

Sounds good. Would be nice to see a test that we can change some private method contents and not recompile the java if we can test that.

Reply:
done